server/controllers/authentication.coffee

```diff
@@ -64,7 +71,8 @@ module.exports.onboarding = (req, res, next) ->
                     # registration mode
                     # TODO: this one is temporary, and need to be removed
                     # when we merge CSS again.
-                    res.render 'index', {env: env, onBoarding: true}
+                    env.currentStep = getCurrentOnboardingStep userData
```

Comment:
It's better to use on line 63, a specific function to compare arrays (this current way doesn't work). I suggest:
```coffee
isArrayEqual = (array1, array2) ->
    if not (array1 and array2)
        return false
    if array1.length isnt array2.length
        return false
    else
        isElementsEqual = array1.every (elem, i) -> elem is array2[i]
        return isElementsEqual
```